modules/core/src/transitions/gpu-interpolation-transition.ts

```diff
@@ -117,18 +123,87 @@ void main(void) {
   gl_Position = vec4(0.0);
 }
 `;
+const vs64 = `\
+#version 300 es
+#define SHADER_NAME interpolation-transition-vertex-shader
+
+uniform float time;
+in ATTRIBUTE_TYPE aFrom;
+in ATTRIBUTE_TYPE aFrom64Low;
+in ATTRIBUTE_TYPE aTo;
+in ATTRIBUTE_TYPE aTo64Low;
+out ATTRIBUTE_TYPE vCurrent;
+out ATTRIBUTE_TYPE vCurrent64Low;
+
+vec2 mix_fp64(vec2 a, vec2 b, float x) {
+  vec2 range = sub_fp64(b, a);
+  return sum_fp64(a, mul_fp64(range, vec2(x, 0.0)));
+}
+
+void main(void) {
+  for (int i=0; i<ATTRIBUTE_SIZE; i++) {
+    vec2 value = mix_fp64(vec2(aFrom[i], aFrom64Low[i]), vec2(aTo[i], aTo64Low[i]), time);
+    vCurrent[i] = value.x;
+    vCurrent64Low[i] = value.y;
+  }
+  gl_Position = vec4(0.0);
+}
+`;
+
+function useFp64(attribute: Attribute): boolean {
+  return attribute.doublePrecision && attribute.value instanceof Float64Array;
+}
 
 function getTransform(device: Device, attribute: Attribute): BufferTransform {
-  const attributeType = getAttributeTypeFromSize(attribute.size);
+  const attributeSize = attribute.size;
+  const attributeType = getAttributeTypeFromSize(attributeSize);
+  const inputFormat = getFloat32VertexFormat(attributeSize);
+  const bufferLayout = attribute.getBufferLayout();
+
+  if (useFp64(attribute)) {
+    return new BufferTransform(device, {
+      vs: vs64,
+      bufferLayout: [
+        {
+          name: 'aFrom',
+          byteStride: 8 * attributeSize,
+          attributes: [
+            {attribute: 'aFrom', format: inputFormat, byteOffset: 0},
+            {attribute: 'aFrom64Low', format: inputFormat, byteOffset: 4 * attributeSize}
+          ]
+        },
+        {
+          name: 'aTo',
+          byteStride: 8 * attributeSize,
+          attributes: [
+            {attribute: 'aTo', format: inputFormat, byteOffset: 0},
+            {attribute: 'aTo64Low', format: inputFormat, byteOffset: 4 * attributeSize}
+          ]
+        }
+      ],
+      modules: [fp64arithmetic],
+      defines: {
+        ATTRIBUTE_TYPE: attributeType,
+        ATTRIBUTE_SIZE: attributeSize
+      },
+      // Default uniforms are not set without this
+      moduleSettings: {},
+      varyings: ['vCurrent', 'vCurrent64Low'],
+      // @ts-expect-error WebGLRenderPipeline only prop TODO - support in RenderPipeline?
+      bufferMode: GL.INTERLEAVED_ATTRIBS,
```

Comment:
@ibgreen This is used byWebGLRenderPipeline but not documented or typed: https://github.com/visgl/luma.gl/blob/050f4e7ac58a3e1117d5ef2da9e1ecb8978aceb3/modules/webgl/src/adapter/resources/webgl-render-pipeline.ts#L67.